### PR TITLE
OTLP-4278 Update Kubernetes example to latest version

### PR DESCRIPTION
**Pull request by shree007**
Description:
The k8s example [here](https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector/main/examples/k8s/otel-config.yaml) has old otel collector image which does not had the new tls config implementation so replacing with new one and command line flag --mem-ballast-size-mib has been deprecated so removed that flag as well.

Link to tracking Issue:
[4247](https://github.com/open-telemetry/opentelemetry-collector/issues/4247)

Testing:
I performed following steps in minikube k8s
$ kubectl apply -f otel-config.yaml

$ kubectl get pod

and checked all pod were up and there were no TLS error

Documentation:
Once it gets merged I will update the docs.

### CODE REVIEW

File: examples/k8s/otel-config.yaml [hunk 1]
```diff
@@ -65,9 +65,7 @@ spec:
       - command:
           - "/otelcol"
           - "--config=/conf/otel-agent-config.yaml"
-          # Memory Ballast size should be max 1/3 to 1/2 of memory.
-          - "--mem-ballast-size-mib=165"
-        image: otel/opentelemetry-collector-dev:latest
+        image: otel/opentelemetry-collector:0.37.1
```
Comment (codeboten) [inline examples/k8s/otel-config.yaml:68]:
```suggestion
        image: otel/opentelemetry-collector:0.38.0
```

File: examples/k8s/otel-config.yaml [hunk 2]
```diff
@@ -177,9 +175,7 @@ spec:
       - command:
           - "/otelcol"
           - "--config=/conf/otel-collector-config.yaml"
-#           Memory Ballast size should be max 1/3 to 1/2 of memory.
-          - "--mem-ballast-size-mib=683"
-        image: otel/opentelemetry-collector-dev:latest
+        image: otel/opentelemetry-collector:0.37.1
```
Comment (codeboten) [inline examples/k8s/otel-config.yaml:178]:
```suggestion
        image: otel/opentelemetry-collector:0.38.0
```

**Comment by codeboten** [comment]
Looks like the CLA check is failing, does this PR superseed https://github.com/open-telemetry/opentelemetry-collector/pull/4278?

**Comment by shree007** [comment]
> Looks like the CLA check is failing, does this PR superseed #4278?

@codeboten yes Easycla is not working, i filed one support ticket for that.
I have tested with 0.38.0 and it's working i fine.
I have replaced previous image tag with 0.38.0 as you commented and resolved the conversation.
Yes #4278 would be suppressed.

**Comment by jpkrohling** [comment]
@nickmango, do you have news about the EasyCLA check?

**Comment by WillsonHG** [comment]
this is a troubleshooting comment /easycla

**Comment by wanyaland** [comment]
> @nickmango, do you have news about the EasyCLA check?

@jpkrohling yes this has been fixed

**Comment by jpkrohling** [comment]
@shree007, looks like there's a missing piece to your local git configuration :-) Look at the link provided in an earlier comment by the EasyCLA integration.

**Comment by shree007** [comment]
> @shree007, looks like there's a missing piece to your local git configuration :-) Look at the link provided in an earlier comment by the EasyCLA integration.

@jpkrohling checking from my end.

**Comment by shree007** [comment]
> > @shree007, looks like there's a missing piece to your local git configuration :-) Look at the link provided in an earlier comment by the EasyCLA integration.
> 
> @jpkrohling checking from my end.

@jpkrohling Looks like Easycla has been resolved, I need your approval  here, please have a look.

**Comment by bogdandrutu** [comment]
@shree007 yaay, thanks a tone for all the effort.

**Review by bogdandrutu — APPROVED** [review]

**Comment by shree007** [comment]
Thanks @jpkrohling  @bogdandrutu @wanyaland for all the help.

**Pull request merged**